superset/views/database/__init__.py

```diff
@@ -16,17 +16,38 @@
 # under the License.
 # pylint: disable=C,R,W
 import inspect
+from typing import Type
 
-from flask import Markup
+from flask import current_app, Markup
 from flask_babel import lazy_gettext as _
+from marshmallow import ValidationError
 from sqlalchemy import MetaData
+from sqlalchemy.engine.url import make_url
+from sqlalchemy.exc import ArgumentError
 
 from superset import security_manager
 from superset.exceptions import SupersetException
 from superset.utils import core as utils
 from superset.views.base import SupersetFilter
 
 
+def sqlalchemy_uri_validator(
+    uri: str, exception: Type[ValidationError] = ValidationError
+) -> None:
+    """
+    Check if a user has submitted a valid SQLAlchemy URI
+    """
+    try:
+        url = make_url(uri.strip())
+    except ArgumentError:
+        raise exception(_("Invalid connnection string could not parse rfc1738 URL"))
```

Comment:
Will this error message be displayed to the user? If so, I suggest making it a bit more user-friendly to those who don't know what RFCs are.

Reply:
Updated